src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -125,10 +135,13 @@ export const useSearch = (services: DiscoverViewServices) => {
   const data$ = useMemo(
     () =>
       new BehaviorSubject<SearchData>({
-        status: shouldSearchOnPageLoad() ? ResultStatus.LOADING : ResultStatus.UNINITIALIZED,
+        status:
+          shouldSearchOnPageLoad() && !skipInitialFetch.current
+            ? ResultStatus.LOADING
+            : ResultStatus.UNINITIALIZED,
         queryStatus: { startTime },
       }),
-    [shouldSearchOnPageLoad, startTime]
+    [shouldSearchOnPageLoad, startTime, skipInitialFetch]
```

Comment:
should we check for `skipInitialFetch.current` instead?

Reply:
+1.